examples/postgres/custom_arrays/README.md

```diff
@@ -0,0 +1,1162 @@
+
+Finally, let’s add another method that returns all services that are online and another one that
+returns all services that are currently offline. The latter is handy to quickly identify a deployment blocker.
+
+### Set Service Online / Offline
+
+Let’s start with a private helper method that sets the Boolean online flag on a service in the database.
+
+```rust 
+    fn set_svc_online(
+        db: &mut Connection,
+        param_service_id: i32,
+        param_online: bool,
+    ) -> QueryResult<()> {
+        match diesel::update(service.filter(service_id.eq(param_service_id)))
+            .set(online.eq(param_online))
+            .returning(Service::as_returning())
+            .get_result(db)
+        {
+            Ok(_) => Ok(()),
+            Err(e) => Err(e),
+        }
+    }
+```
+
+Here the generated DSL really shines as you can easily select a field and set it to a new value. Notice, the return type
+is void in case of success. Realize, if you were to return the updated Boolean flag, you would cause a non-trivial
+ambiguity on the call-site. For setting the Boolean flag true, the result would be true, which is correct, but there is
+also a risk of the return value being interpreted as the operation having succeeded. And that matters, because if you
+set the flag to false, the same method return false, which is also correct, but could be misunderstood as the operation
+didn’t completed because false was returned. A good API is mostly free of ambiguity and therefore, therefore you just
+return Ok(()) in case of success because there is no way to misunderstand a returned ok.
+
+Now, let’s add two public wrapper methods that set the flag to either true or false.
+
+```rust 
+    pub fn set_service_online(db: &mut Connection, param_service_id: i32) -> QueryResult<()> {
+        Self::set_svc_online(db, param_service_id, true)
+    }
+
+    pub fn set_service_offline(db: &mut Connection, param_service_id: i32) -> QueryResult<()> {
+        Self::set_svc_online(db, param_service_id, false)
+    }
+```
+
+### Check Service Exists
+
+Next up, we add a method to check if service is in the database and the easiest way to implement this is to test if find
+returns a result or error.
+
+```rust
+    pub fn check_if_service_id_exists(
+        db: &mut Connection,
+        param_service_id: i32,
+    ) -> QueryResult<bool> {
+        match service.find(param_service_id).first::<Service>(db) {
+            Ok(_) => Ok(true),
+            Err(_) => Ok(false),
+        }
+    }
+```
+
+### Check Service Online
+
+To check if a service is online, we again lean on the generated DSL:
+
+```rust
+    pub fn check_if_service_id_exists(
+        db: &mut Connection,
+        param_service_id: i32,
+    ) -> QueryResult<bool> {
+        match service.find(param_service_id).first::<Service>(db) {
+            Ok(_) => Ok(true),
+            Err(_) => Ok(false),
+        }
+    }
```

Comment:
That seems to be copy + pasted from above. We should likely use the correct function from the example here